sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -343,8 +343,14 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]]
             }
         }
 
-        val (planAfterRule, newAttrMapping) = CurrentOrigin.withOrigin(origin) {
-          rule.applyOrElse(newPlan, (plan: PlanType) => plan -> Nil)
+        val (planAfterRule, newAttrMapping) = {
+          if (rule.isDefinedAt(newPlan)) {
+            CurrentOrigin.withOrigin(origin) {
+              rule.apply(newPlan)
+            }
+          } else {
+            plan -> Nil
```

Comment:
This should be `newPlan -> Nil`; fixed in 2cd4d9deaa1b08c89a513ab6ba9b61481884e754